src/transformers/modeling_utils.py

```diff
@@ -1713,19 +1800,26 @@ def tie_encoder_to_decoder_recursively(
                         decoder_modules[decoder_name],
                         encoder_modules[encoder_name],
                         module_name + "/" + name,
+                        base_encoder_name,
                         uninitialized_encoder_weights,
                         depth=depth + 1,
+                        total_encoder_name=f"{total_encoder_name}.{encoder_name}",
+                        total_decoder_name=f"{total_decoder_name}.{decoder_name}",
```

Comment:
Here - do we want to account for when the string is empty? 

```suggestion
                        total_encoder_name=f"{total_encoder_name}.{encoder_name}" if total_encoder_name else encoder_name,
                        total_decoder_name=f"{total_decoder_name}.{decoder_name}" if total_decoder_name else decoder_name,
```